memtx: fix heap-use-after-free of tuple stories caused by space alter (#8820)
tarantool/tarantool: changelogs/unreleased/gh-8781-heap-use-after-free-in-memtx-tx-delete-gap.md

## bugfix/memtx

> * Fixed a heap-use-after-free bug in transaction manager which could occur
>   during a DDL operation happening concurrently with a transaction (gh-8781).

Reviewer: ```suggestion
* Fixed a heap-use-after-free bug in the transaction manager, which could occur
  when performing a DDL operation concurrently with a transaction (gh-8781).
```

Author: Fixed.
